src/Functional.CQS.AOP.Caching.Infrastructure.DistributedCache.Redis/RedisClientExtensions.cs

```diff
@@ -78,7 +84,21 @@ internal static class RedisClientExtensions
 			});
 		}
 
-		private static Result<RedisText, Exception> ExecuteLuaShaSafely(this IRedisClient client, string sha1, string[] keys, string[] args) => Result.Try(() => client.ExecLuaSha(sha1, keys, args));
-		private static Result<IEnumerable<string>, Exception> ExecuteLuaShaAsListSafely(this IRedisClient client, string sha1, string[] keys, string[] args) => Result.Try(() => client.ExecLuaShaAsList(sha1, keys, args).AsEnumerable());
+		private static Result<RedisValue, Exception> ExecuteLuaShaSafely(this RedisClient client, LoadedLuaScript script, string[] keys, string[] args) => Result.Try(() => (RedisValue)client.ExecuteScript(script, keys, args));
+		private static Result<string[], Exception> ExecuteLuaShaAsListSafely(this RedisClient client, LoadedLuaScript script, string[] keys, string[] args) => Result.Try(() => (string[])client.ExecuteScript(script, keys, args));
+
+		private static T ExecCachedLua<T>(this RedisClient client, string script, Func<LoadedLuaScript, T> factory)
```

Comment:
A new extension method (using a `ServiceStack.Redis` method name) so I did not have make as many changes to existing code.